app/src/main/java/com/swent/assos/ui/screens/assoDetails/NewsDetails.kt

```diff
@@ -1,7 +1,146 @@
 package com.swent.assos.ui.screens.assoDetails
 
+import android.net.Uri
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.semantics.testTagsAsResourceId
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import coil.compose.AsyncImage
+import coil.compose.rememberAsyncImagePainter
+import com.swent.assos.R
+import com.swent.assos.model.data.News
 import com.swent.assos.model.navigation.NavigationActions
+import com.swent.assos.model.view.NewsViewModel
+import com.swent.assos.ui.components.PageTitleWithGoBack
+import java.time.LocalDateTime
 
-@Composable fun NewsDetails(newsId: String, navigationActions: NavigationActions) {}
+@OptIn(ExperimentalComposeUiApi::class, ExperimentalMaterial3Api::class)
+@Composable
+fun NewsDetails(newsId: String, navigationActions: NavigationActions) {
+
+  val viewModel: NewsViewModel = hiltViewModel()
+
+  val news by viewModel.allNews.collectAsState()
+
+  val specificNews =
+      news.find { it.id == newsId }
+          ?: News(
+              id = "0000",
+              title = "NO TITLE",
+              description = "NO DESCRIPTION",
+              images = listOf(Uri.EMPTY),
+              createdAt = LocalDateTime.now(),
+              associationId = "",
+              eventIds = mutableListOf())
```

Comment:
I suggest you create a method named getNewsFromId so that you don't have to collect all the news items and then search through them to find the one with the correct ID.